#!/usr/bin/env python
PACKAGE = "plugin_manager"

import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()


#matchers_enum = gen.enum([ gen.const("FLANN", int_t, 0, "FLANN"),
# gen.const("Bruteforce_Hamming_1", int_t, 1, "Bruteforce_Hamming_1"),
# gen.const("Bruteforce_Hamming_2", int_t, 2, "Bruteforce_Hamming_2"),
# gen.const("Bruteforce_L1", int_t, 3, "Bruteforce_L1"),
# gen.const("Bruteforce_L2", int_t, 4, "Bruteforce_L2"),],
# "Extractor Matchers")
#gen.add("descriptor_matcher", int_t, 0, "Descriptor matcher", 0, 0, 1, edit_method=matchers_enum)
#gen.add("feature_detector_parameter_name", str_t, 0, "Enter the name of a algorithm parameter to change", "nFeatures")
#gen.add("max_radius_search_dist", double_t, 0, "radius search threshold", .5, 0, 2000)
#gen.add("horizontal_matches", bool_t, 0, "Horizontal matches", False)
#gen.add("horizontal_threshold", int_t, 0, "Horizontal matches threshold", 1, 0, 100)

size_enum = gen.enum([ gen.const("INIT", int_t, 0, "Init mode"),
                  gen.const("ALL", int_t, 1, "Estimating the push point"),
                  gen.const("PUSH_POINT", int_t, 2, "Estimating the push point"),
                  gen.const("STATIC_SEGMENTATION", int_t, 3, "Static segmentation") ],
                  "Selected mode")

gen.add("manager_mode", int_t, 0, "A mode that the program is in", 1, 0, 3, edit_method=size_enum)


size_enum = gen.enum([ gen.const("PUSH_POINT_HAUSMAN", int_t, 0, "PUSH_POINT_HAUSMAN"),
                  gen.const("PUSH_POINT_IMPL2", int_t, 1, "PUSH_POINT_IMPL2"),
                  gen.const("PUSH_POINT_IMPL3", int_t, 2, "PUSH_POINT_IMPL3") ],
                  "Selected mode")

gen.add("push_point_mode", int_t, 0, "A push point mode is: ", 1, 0, 2, edit_method=size_enum)

size_enum = gen.enum([ gen.const("STATIC_SEGMENTATION_HAUSMAN", int_t, 0, "STATIC_SEGMENTATION_HAUSMAN"),
                  gen.const("STATIC_SEGMENTATION_IMPL2", int_t, 1, "STATIC_SEGMENTATION_IMPL2"),
                  gen.const("STATIC_SEGMENTATION_IMPL3", int_t, 2, "STATIC_SEGMENTATION_IMPL3") ],
                  "Selected mode")

gen.add("static_segmentation_mode", int_t, 0, "A static segmentation mode is: ", 1, 0, 2, edit_method=size_enum)


#gen.add("estimate_pushpoint", bool_t, 0, "estimate_pushpoint", False)
#gen.add("static_segment", bool_t, 0, "static_segment", False)


exit(gen.generate(PACKAGE, "plugin_manager", "PluginManager"))
